src/diffusers/models/attention_processor.py

```diff
@@ -369,7 +369,20 @@ def set_use_memory_efficient_attention_xformers(
                 )
                 processor = XFormersAttnAddedKVProcessor(attention_op=attention_op)
             else:
-                processor = XFormersAttnProcessor(attention_op=attention_op)
+                processor = self.processor
+                if isinstance(self.processor, (IPAdapterAttnProcessor, IPAdapterAttnProcessor2_0)):
+                    processor = IPAdapterXFormersAttnProcessor(hidden_size=self.processor.hidden_size,
+                        cross_attention_dim=self.processor.cross_attention_dim,
+                        scale=self.processor.scale,
+                        num_tokens=self.processor.num_tokens,
+                        attention_op=attention_op)
+                    processor.load_state_dict(self.processor.state_dict())
```

Comment:
why do we need to `load_state_dict` again here?

Reply:
Well, I couldn't identify why, but if I don't reload the state_dict again here after assigning the new class, the final result in the image is not applied, it's as if the model was not being used. See a final image when there is no state dict and another when there is. So I noticed that something similar is done in custom diffusion and I decided to do the same. If you have a better solution I would like to try it.

Without load_sate_dict:
![result_1_diff](https://github.com/user-attachments/assets/38d12f47-5bdf-4bd6-bc0a-71b3d953cf83)

With load_state_dict:
![result_1_diff](https://github.com/user-attachments/assets/deab48f7-5f7e-49b9-9be6-4083a435cfc2)

Reply:
oh yeah it comes with weights https://github.com/huggingface/diffusers/blob/1b392544c758e45cc7097cc35309cb8cc11798e4/src/diffusers/models/attention_processor.py#L4351

(i had forgotten about that  sorry! lol)